lib/forwardanalyzer.cpp

```diff
@@ -646,6 +646,9 @@ namespace {
                     }
                 } else if (tok->isControlFlowKeyword() && Token::Match(tok, "if|while|for (") &&
                            Token::simpleMatch(tok->next()->link(), ") {")) {
+                    const int branchCount = analyzer->countBranch();
+                    if (settings.checkLevel == Settings::CheckLevel::normal && branchCount > 4)
```

Comment:
We should generate a message that informs the user of this so it is clear that the exhaustive mode needs to be used for the full analysis.

Reply:
We should also make the threshold configurable.

Reply:
yes